src/main/java/com/suse/salt/netapi/calls/LocalCall.java

```diff
@@ -38,50 +38,49 @@
  *
  * @param <R> the return type of the called function
  */
-public class LocalCall<R> implements Call<R> {
+public class LocalCall<R> extends AbstractCall<R> {
 
-    private final String functionName;
     private final Optional<List<?>> arg;
-    private final Optional<Map<String, ?>> kwarg;
-    private final TypeToken<R> returnType;
     private final Optional<?> metadata;
 
     public LocalCall(String functionName, Optional<List<?>> arg,
-            Optional<Map<String, ?>> kwarg, TypeToken<R> returnType,
+            Optional<Map<String, ?>> kwargs, TypeToken<R> returnType,
             Optional<?> metadata) {
-        this.functionName = functionName;
+        super(functionName, kwargs, returnType);
+        this.arg = arg;
+        this.metadata = metadata;
+    }
+
+    public LocalCall(String moduleName, String functionName, Optional<List<?>> arg,
+                     Optional<Map<String, ?>> kwargs, TypeToken<R> returnType,
+                     Optional<?> metadata) {
+        super(moduleName, functionName, kwargs, returnType);
         this.arg = arg;
-        this.kwarg = kwarg;
-        this.returnType = returnType;
         this.metadata = metadata;
     }
 
     public LocalCall(String functionName, Optional<List<?>> arg,
-            Optional<Map<String, ?>> kwarg, TypeToken<R> returnType) {
-        this(functionName, arg, kwarg, returnType, Optional.empty());
+            Optional<Map<String, ?>> kwargs, TypeToken<R> returnType) {
+        this(functionName, arg, kwargs, returnType, Optional.empty());
     }
 
     public LocalCall<R> withMetadata(Object metadata) {
-        return new LocalCall<>(functionName, arg, kwarg, returnType, Optional.of(metadata));
+        return new LocalCall<>(getFunctionName(), arg, getKwargs(), getReturnType(),
+                Optional.of(metadata));
     }
 
     public LocalCall<R> withoutMetadata() {
-        return new LocalCall<>(functionName, arg, kwarg, returnType, Optional.empty());
-    }
-
-    public TypeToken<R> getReturnType() {
-        return returnType;
+        return new LocalCall<>(getFunctionName(), arg, getKwargs(), getReturnType(),
+                Optional.empty());
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public Map<String, Object> getPayload() {
-        HashMap<String, Object> payload = new HashMap<>();
-        payload.put("fun", functionName);
+        Map<String, Object> payload = super.getPayload();
         arg.ifPresent(arg -> payload.put("arg", arg));
-        kwarg.ifPresent(kwarg -> payload.put("kwarg", kwarg));
```

Comment:
Thanks for the feedback and sorry for the inconvenience caused by my change. Somehow tests gave me wrong impression that my code didn't break anything, I should have been more careful.
 
I will make changes suggested above through a new PR.